Bonsai.Configuration/PackageConfigurationUpdater.cs

```diff
@@ -69,6 +69,36 @@ string GetRelativePath(string path)
             return PathUtility.GetPathWithDirectorySeparator(relativeUri.ToString());
         }
 
+        static string CombinePath(string path1, string path2)
+        {
+            return PathUtility.GetPathWithForwardSlashes(Path.Combine(path1, path2));
+        }
+
+        static PackageConfiguration NormalizePathSeparators(PackageConfiguration configuration)
+        {
+            var normalized = new PackageConfiguration();
+            normalized.ConfigurationFile = configuration.ConfigurationFile;
+            normalized.Packages.AddRange(configuration.Packages);
+            normalized.AssemblyReferences.AddRange(configuration.AssemblyReferences);
+
+            foreach (var assemblyLocation in configuration.AssemblyLocations)
+            {
+                normalized.AssemblyLocations.Add(
+                    assemblyLocation.AssemblyName,
+                    assemblyLocation.ProcessorArchitecture,
+                    PathUtility.GetPathWithForwardSlashes(assemblyLocation.Location));
+            }
+
+            foreach (var folder in configuration.LibraryFolders)
+            {
+                normalized.LibraryFolders.Add(
+                    PathUtility.GetPathWithForwardSlashes(folder.Path),
+                    folder.Platform);
+            }
+
```

Comment:
It doesn't appear to actually matter, but `ExtensionFolders` should get copied+normalized too. (It should at least be copied over.)

Reply:
Same reasoning as above for `ConfigurationFile`. In addition the reason for not copying `ExtensionFolders` is that they are explicitly out of scope of the `PackageConfigurationUpdater` and should never be touched by it or risk saving anything to file.

Reply:
I was giving it another look and it might be the case that we can't actually copy the `PackageConfiguration` object at all, but rather try to modify it in place. The reason is that the original object gets passed over in sequence to different asynchronous package recovery updaters and they were supposed to fill it with restored packages, whereas now this will never happen.

Reply:
Just to clarify: You're saying you were originally trying to avoid modifying the existing `PackageConfiguration` but now you're OK with it? (I agree it seems like it should be fine, and the revised implementation indirectly addresses my concerns with this getting out of sync with `PackageConfiguration`.)

Reply:
Yup, that's right. One of those cases where it seemed like potentially a good idea but actually isn't 😅 (in the current code base at least). We can revisit all this later but out of scope of this patch.